luigi/contrib/slack_notifications.py

```diff
@@ -0,0 +1,181 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+''' Supports sending slack notifications for different task events.
+
+Usage:
+
+Create an incomming webhook on slack: https://api.slack.com/incoming-webhooks
+
+Setup the slack integration with the following code.
+This can be done in the same module where the luigi tasks are defined.
+
+.. code-block:: python
+
+    from luigi.contrib.slack_notifications import configure_slack
+    configure_slack()
+
+Use a configuration like the following on luigi.cfg
+
+.. code-block:: ini
+
+    [slack]
+    webhook = https://hooks.slack.com/services/something
+    events = ["START", "SUCCESS", "FAILURE"]
+    template = Luigi task {task_id} event {event}: {args} {kwargs}
+
+
+Alternatives:
+
+Another plugin with similar purpouse is luigi-slack (https://github.com/bonzanini/luigi-slack). The major differences between this and luigi-slack is that:
+
+- Inspired by email notifications (notifications.py): simple configuration though luigi.cfg (no need for python configuration)
+- Minimal changes to python code (only call configure_slack()), no need create a new wrapper around luigi.run.
```

Comment:
> would you consider creating a new repo like github.com/spotify/luigi-slack-requests (-requests to distinguish it from the existing repo)?

Maybe. We already have an internal repo with other tools (where users install that package and get slack + other things). The goal to PR here is to make it easily available to others. I work on setting up a new repo, it will take a few more days to do that.